porcupine/plugins/hover.py

```diff
@@ -16,24 +16,27 @@ class Response(utils.EventDataclass):
 class HoverManager:
     def __init__(self, textwidget: tkinter.Text):
         self._textwidget = textwidget
-        self._label: tkinter.Label | None = None
-        self._location = textwidget.index("insert")
-
-    def hide_label(self, junk: object = None) -> None:
-        if self._label is not None:
-            self._label.destroy()
-            self._label = None
-
-    def _show_label(self, message: str) -> None:
-        self.hide_label()
         self._label = tkinter.Label(
             self._textwidget,
-            text=message,
             justify="left",
             # some hack to get a fitting popup bg
             bg=utils.mix_colors(self._textwidget["bg"], self._textwidget["fg"], 0.8),
             fg=self._textwidget["fg"],
         )
+        self._location = textwidget.index("insert")
+
+    def hide_label(self, junk: object = None) -> None:
+        self._label.place_forget()
+
+    def _show_label(self, message: str) -> None:
+        self.hide_label()
+
+        if message != self._label["text"]:
+            self._label.configure(
+                text=message, wraplength=1000  # place_popup will adjust the wraplength
+            )
+            self._label.after(500, None)
```

Comment:
This is equivalent to `time.sleep(0.5)`: it freezes the GUI for 0.5 seconds. This is easy to see if you type something while you're hovering, for example, but I think I would occasionally notice it when using the editor normally. You need to make a callback function and pass it to the `.after()` method.

This way to use the `.after()` method is just dumb and weird. Here's its source code:

```python3
    def after(self, ms, func=None, *args):
        """..."""
        if not func:
            # I'd rather use time.sleep(ms*0.001)
            self.tk.call('after', ms)
            return None
        else:
            ...
```

Whoever wrote this a long time ago wanted to mirror what Tcl's `after` command does (there's no separate `sleep` in Tcl), and left a comment about how it felt weird in Python.

Reply:
Presumably you should use `after_cancel()` too: if another hover appears or `hide_label()` is called during the 500 milliseconds, we don't want to do anything for this hover.

Reply:
Yea, lol I'm super stupid, and I even remembered that comment. Tcl is weird.